### DISCUSSION TITLE
Move AdvancedCollectionView to CommunityToolkit.Mvvm.Collections

### DISCUSSION THREAD

**Discussion by e0ff** [CommunityToolkit/dotnet#809] [General]
There doesn't seem to be a reason why AdvcancedCollectonView couldn't be used on other platforms such as WPF and Avalonia.

**Comment by michael-hawker** [comment]
Thank @jhert0 for opening this in Labs, moving to the .NET Community Toolkit repo as it's about migrating a feature from the Windows Community Toolkit to it.

Think this would be dependent on moving the WeakEventListener too, as the ACV uses that feature as well. So, it's related to this open issue as well as a prerequisite: https://github.com/CommunityToolkit/dotnet/issues/404

Doing a quick pass otherwise, it does seem like everything is based on just the .NET interfaces. ACV is missing a couple of features, but didn't think about having a generalized version that's UI agnostic. Wonder if we could grab those improvements from WPF itself?

**Comment by michael-hawker** [comment]
Ah, @Arlodotexe pointed out (I missed it as we have global namespaces in WCT), that the [ISupportIncrementalLoading](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.data.isupportincrementalloading) interface is Windows only and of course the underlying [ICollectionView](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.data.icollectionview) is of course one too.

So, there's no common base that could be shared here?

**Reply by e0ff** [reply]
I missed those interfaces being defined in windows only namespaces. I'm not sure what to do for ISupportIncrementalLoading but would it work to have an extensions method on AdvancedCollectionView that could be called something like AsICollectionView and cast the AdvancedCollectionView to that interface? The only problem I see is that you would have to update the AdvancedCollectionView if the ICollectionView interface changes but I'm not sure if ever changes.
